### ISSUE TITLE
Proper user agent for Client Login flow

### ISSUE THREAD

**Issue by rullzer** [nextcloud/server#8077]
As noted in https://github.com/nextcloud/server/pull/7754 we have to send more detailed UserAgents when regsitering a new device using the client login flow.

Basically there are 4 tasks:

1. Come up with a user agent string we need that contains all the info
2. Send this string properly from the clients
  * Android
  * IOS
  * Desktop
3. Make sure we also parse it in https://github.com/nextcloud/server/blob/master/core/Controller/ClientFlowLoginController.php#L117 so we show a human readable string
4. Make sure to parse it on the Secuirity (personal) page so we show it properly.

@ChristophWurst as discussed.

**Comment by mario** [comment]
For Talk, we now show: Device name (Nextcloud Talk) on both iOS and Android. For Nextcloud Android app we show: Device name.

**Comment by rullzer** [comment]
@mario ok I think that is fine

But maybe it would be even better to do "Device name (Nextcloud Talk on Android)" or something but pff ;)

Also, (an idea!), we could in the webui show a nice nextcloud talk icon when it is the talk that is connected. Just to make it more visual. I'll create a ticket about that as well.

**Comment by mario** [comment]
@rullzer the issue is that this appears on the web login page, so that would make it really long, unless the server changes NOT to show a part of it :P I like the idea of showing a Talk icon! :)

@tobiasKaminsky I suggest we do similar thing for the Nc Android app, like we talked, so maybe:
"Device name (Nextcloud Files)"? :)

**Comment by tobiasKaminsky** [comment]
> (Nextcloud Talk on Android)"

As the device name is already there, I think Android/iOS is not needed.

> "Device name (Nextcloud Files)

There was some discussion to keep the core system as "nextcloud", but I do not 100% remember this.

**Comment by mario** [comment]
@tobiasKaminsky you can always call your iPhone "Android", and also it's hard to parse device name to know that it's Android.

**Comment by MorrisJobke** [comment]
What are the next steps here?

**Comment by rullzer** [comment]
Not something for 14 anymore. Probably still good to do but it also requires a proper format etc.

**Comment by tobiasKaminsky** [comment]
As changing the user agent killed Global Scale once, we have to make sure this does not happen again.

**Comment by szaimen** [comment]
I suppose this issue is still valid? If not, please close this issue!

**Comment by michal25** [comment]
Is it here some possibility to disable useragents with defined string. Such as "Windows"?

The idea is to disable user agents from selected operating systems.

**Comment by tobiasKaminsky** [comment]
This is only about transparency, not about blocking users/clients based on OS.
Please discuss this in a separate issue.
